### PR TITLE
Setup data upgrade project

### PR DESCRIPTION
**Pull request by JakeGinnivan**

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "semi": false,
+    "trailingComma": "all",
+    "singleQuote": true
```
Comment (antonjb) [inline .prettierrc:4]: should this have similar settings to editorconfig?

File: readme.md [hunk 2]
```diff
@@ -1 +1,192 @@
 # Nanobots data upgrader
+
+Nanobots is a library which helps you fix data and refactor your database in a safe, reliable way!
+
+## Problems with breaking schema changes
+
+There are a few problems with making a breaking schema change (like a rename), and many teams/companies just choose not to make them. This means your database cannot be refactored and evolved like you code, which can cause problems downt the track.
```
Comment (antonjb) [inline readme.md]: Bit of grammar weirdness in here but the one that you might want to change is `...like you code` and `downt`

File: src/applied-upgrades-utils.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+export function addAppliedUpgrades(appliedUpgrades: string[] | null, upgradeName: string) {
+    if (appliedUpgrades) {
+        if (appliedUpgrades.includes(upgradeName)) {
+            return appliedUpgrades
+        }
+
+        return [...appliedUpgrades, upgradeName]
+    }
+
+    return [upgradeName]
```
Comment (antonjb) [inline src/applied-upgrades-utils.ts]: Could this be slightly clearer if you use a default value for appliedUpgrades?

```ts
export function addAppliedUpgrades(appliedUpgrades: string[] = [], upgradeName: string) {
        if (appliedUpgrades.includes(upgradeName)) {
            return appliedUpgrades
        }

        return [...appliedUpgrades, upgradeName]
}
```

File: .prettierrc (hunk 1 above)
Comment (JakeGinnivan) [reply .prettierrc:4]: Possible, it seems to work though, I wonder if it picks it up. You mean things like tab width?

File: readme.md (hunk 2 above)
Comment (JakeGinnivan) [reply readme.md]: Making a few small changes

File: src/applied-upgrades-utils.ts (hunk 3 above)
Comment (JakeGinnivan) [reply src/applied-upgrades-utils.ts]: Looking at the consuming code, it handles the null case already. Simplified

**Review by antonjb — APPROVED** [review]

**Review by andyjdavis — APPROVED** [review]

**Pull request merged**